config/assert_test.go

```diff
@@ -0,0 +1,209 @@
+// Copyright 2022 Mineiros GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config_test
+
+import (
+	"fmt"
+	"testing"
+
+	hhcl "github.com/hashicorp/hcl/v2"
+	"github.com/madlambda/spells/assert"
+	"github.com/mineiros-io/terramate/config"
+	"github.com/mineiros-io/terramate/errors"
+	"github.com/mineiros-io/terramate/hcl"
+	"github.com/mineiros-io/terramate/hcl/eval"
+	"github.com/mineiros-io/terramate/test"
+	"github.com/zclconf/go-cty/cty"
+)
+
+func TestAssertConfigEval(t *testing.T) {
+	type testcase struct {
+		name       string
+		assert     hcl.AssertConfig
+		namespaces namespaces
+		want       config.Assert
+		wantErr    error
+	}
+
+	expr := func(s string) hhcl.Expression {
+		return test.NewExpr(t, s)
+	}
+
+	tcases := []testcase{
+		{
+			name: "using literals",
+			assert: hcl.AssertConfig{
+				Assertion: expr(`"a" == "terramate"`),
+				Message:   expr(`"something"`),
+			},
+			want: config.Assert{
+				Assertion: false,
+				Message:   "something",
+			},
+		},
+		{
+			name: "accessing namespace values",
+			namespaces: namespaces{
```

Comment:
this is cool :heart:

Reply:
a lot of places could be tested independently of the stack metadata or globals.. what matters is just supports whatever is in the context.